src/extension/ln/nostr/index.ts

```diff
@@ -41,6 +56,8 @@ export default class NostrProvider {
   // NOTE: new call `action`s must be specified also in the content script
   execute<T>(action: string, args?: Record<string, unknown>): Promise<T> {
     return new Promise((resolve, reject) => {
+      const id = Math.random().toString().slice(4);
+      window.nostr._requests[id] = { resolve, reject };
```

Comment:
I don't believe the _requests object is actually needed. Execute creates a dynamic function for this specific request, and can compare messageEvent.data.id against the id that was passed in the postMessage.

Reply:
I think it's needed because the promise isn't resolved directly but through another callback from the global `message` event posted on the window. 

[window.addEventListener("message", handleWindowMessage);](https://github.com/getAlby/lightning-browser-extension/blob/fix/race-condition/src/extension/ln/nostr/index.ts#L107)

Since multiple calls can be executed in parallel and the order of events doesn't have to match the order they were called that lead to the race conditions that could lead to certain functions returning results of other function calls. 

`window.nostr.signMessage` could potentially return the result of `window.nostr.getPublicKey` under certain conditions. By referencing the promises through this unique id that shouldn't be the case anymore.

Reply:
Thanks for your feedback, with the added check for the id in the event handler this became much simpler 👌